Exiled.Events/Patches/Events/Map/AnnouncingNtfEntrance.cs

```diff
@@ -25,18 +29,92 @@ namespace Exiled.Events.Patches.Events.Map
-            string[] unitInformation = regular.Split('-');
+            List<CodeInstruction> newInstructions = ListPool<CodeInstruction>.Shared.Rent(instructions);
+
+            LocalBuilder ev = generator.DeclareLocal(typeof(AnnouncingNtfEntranceEventArgs));
+            LocalBuilder unitInformation = generator.DeclareLocal(typeof(string[]));
+
+            Label ret = generator.DefineLabel();
+
+            const int offset = 1;
+            int index = newInstructions.FindIndex(instruction => instruction.opcode == OpCodes.Stloc_1) + offset;
+
+            // int scpsLeft = ReferenceHub.GetAllHubs().Values.Count(x => x.characterClassManager.CurRole.team == Team.SCP && x.characterClassManager.CurClass != RoleType.Scp0492);
+            // string[] unitInformation = unitName.Split('-');
+            // var ev = new AnnouncingNtfEntranceEventArgs(scpsLeft, unitInformation[0], int.Parse(unitInformation[1]));
+            // Map.OnAnnouncingNtfEntrance(ev);
+            // if(!ev.IsAllowed)
+            //     return;
+            // unitName = $"{ev.UnitName}-{ev.UnitNumber};
+            // cassieUnitName = this.GetCassieUnitName(unitName);
+            // scpsLeft = ev.ScpsLeft;
+            newInstructions.InsertRange(index, new[]
+            {
+                // int scpsLeft = ReferenceHub.GetAllHubs().Values.Count(x => x.characterClassManager.CurRole.team == Team.SCP && x.characterClassManager.CurClass != RoleType.Scp0492);
+                new CodeInstruction(OpCodes.Ldloc_1),
+
+                // string[] unitInformation = unitName.Split('-');
+                new(OpCodes.Ldarg_1),
+                new(OpCodes.Ldc_I4_1),
+                new(OpCodes.Newarr, typeof(char)),
+                new(OpCodes.Dup),
+                new(OpCodes.Ldc_I4_0),
+                new(OpCodes.Ldc_I4_S, 45),
+                new(OpCodes.Stelem_I2),
+                new(OpCodes.Call, Method(typeof(string), nameof(string.Split), new[] { typeof(char[]) })),
+
+                // var ev = new AnnouncingNtfEntranceEventArgs(scpsLeft, unitInformation[0], int.Parse(unitInformation[1]));
+                // Map.OnAnnouncingNtfEntrance(ev);
+                new(OpCodes.Dup),
+                new(OpCodes.Stloc_S, unitInformation.LocalIndex),
+                new(OpCodes.Ldc_I4_0),
+                new(OpCodes.Ldelem_Ref),
+                new(OpCodes.Ldloc_S, unitInformation.LocalIndex),
+                new(OpCodes.Ldc_I4_1),
+                new(OpCodes.Ldelem_Ref),
+                new(OpCodes.Call, Method(typeof(int), nameof(int.Parse), new[] { typeof(string) })),
+                new(OpCodes.Ldc_I4_1),
+                new(OpCodes.Newobj, GetDeclaredConstructors(typeof(AnnouncingNtfEntranceEventArgs))[0]),
+                new(OpCodes.Dup),
+                new(OpCodes.Dup),
+                new(OpCodes.Stloc_S, ev.LocalIndex),
+                new(OpCodes.Call, Method(typeof(Map), nameof(Map.OnAnnouncingNtfEntrance))),
```

Comment:
I think this is my mistake in logic, I don't like accessing the `unitInformation` here, but I don't have any ideas how to improve it